lib/hyrax/acl/access_control_list.rb

```diff
@@ -15,17 +11,67 @@ module Acl
     # `Hyrax::Resource.
     #
     class AccessControlList
+      DISCOVER = :discover
+      EDIT     = :edit
+      READ     = :read
+
       ##
-      # @!attribute [r] access_control
-      #   @return [Hyax:Acl:AccessControl]
-      attr_reader :access_control
+      # @!attribute [rw] resource
+      #   @return [Valkyrie::Resource]
+      # @!attribute [r] persister
+      #   @return [#save]
+      # @!attribute [r] query_service
+      #   @return [#find_inverse_references_by]
+      attr_reader :persister, :query_service
+      attr_accessor :resource
+
+      ##
+      # @param resource [Valkyrie::Resource]
+      # @param persister [#save] defaults to the configured Hyrax persister
+      # @param query_service [#find_inverse_references_by] defaults to the
+      #   configured Hyrax query service
+      def initialize(resource:, persister:, query_service:)
+        self.resource  = resource
+        @persister     = persister
+        @query_service = query_service
+      end
 
       ##
       # @api public
       #
-      # @param :access_control [Hyax:Acl:AccessControl]
-      def initialize(access_control:)
-        @access_control = access_control
+      # @param permission [Hyrax::Permission]
+      #
+      # @return [Boolean]
+      def <<(permission)
+        permission.access_to = resource.id
```

Comment:
i'm uncertain off the top of my head (and couldn't quickly discover with project-wide grep) whether Hyrax has existing code that depends on this. the fundamental use case is: i have user input that results in domain model objects `Hyrax::Permission` and i'd like to set them directly on an object. `acl = Hyrax::Acl::AccessControlList(resource); acl << permissions; acl.save`

the view in development was that the Grant/Revoke DSL was a nicety provided over this lower level service. my experience has been that closing off direct access to low level APIs leads to users reaching through into private ones. since this API is fairly simple, and felt like a prerequisite to implementation for Grant/Revoke, i left it public.

Reply:
the difference between `#<<` and `#delete` is arguably logical:

i never want to add a `Permission` with another target object, so `#<<` casts (maybe it should error instead? i think i felt at the time that adding with an empty target was likely to be the only situation to actually arise, and that the intention in this case would uniformly be to add the permission with the target for this ACL. this is probably not in use, and i think we could safely reconsider.

on the flip side, i don't want `acl << public_read_my_resource ; acl.delete(public_read_other_resource)` to result in an empty permission set.